src/OpenApi/src/Services/OpenApiOptions.cs

```diff
@@ -14,6 +14,7 @@ namespace Microsoft.AspNetCore.OpenApi;
 public sealed class OpenApiOptions
 {
     internal readonly List<IOpenApiDocumentTransformer> DocumentTransformers = [];
+    internal readonly List<Func<OpenApiSchema, OpenApiSchemaTransformerContext, CancellationToken, Task>> SchemaTransformers = [];
```

Comment:
What are scenarios where someone could have an async schema transformer?

Reply:
Is XML comments file from disk using async and then applying changes to schema an example?

Reply:
XML comments support will be handled separately in this implementation, although theoretically there's nothing stopping someone from rolling out an implementation on their own or reading additional metadata from some non-XML file.